### PR TITLE
[RFR] display board

### PR DESCRIPTION
**Pull request by mchaffotte**
adding tests

### CODE REVIEW

File: quoridor-app/app/api/game_api.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const API_PATH = "http://192.168.86.56:8383/games";
+const JSON_TYPE = "application/json";
+
+export class GameAPI {
```
Comment (zyhou) [inline quoridor-app/app/api/game_api.js]: No need class

File: quoridor-app/app/api/game_api.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+const API_PATH = "http://192.168.86.56:8383/games";
```
Comment (zyhou) [inline quoridor-app/app/api/game_api.js]: config file ?

File: quoridor-app/app/api/game_api.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+export class GameAPI {
+  async createGame() {
+    const res = await fetch(API_PATH, {
+      method: "POST",
+      headers: {
+        Accept: JSON_TYPE,
+        "Content-Type": JSON_TYPE
+      },
+      body: JSON.stringify({
+        boardSize: 9,
```
Comment (zyhou) [inline quoridor-app/app/api/game_api.js]: constant

File: quoridor-app/app/api/game_api.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+const API_PATH = "http://192.168.86.56:8383/games";
+const JSON_TYPE = "application/json";
+
+export class GameAPI {
+  async createGame() {
+    const res = await fetch(API_PATH, {
```
Comment (zyhou) [inline quoridor-app/app/api/game_api.js]: Rename this variable `resultat` or `game` or etc..

File: quoridor-app/app/api/game_api.js (hunk 2 above)
Comment (zyhou) [inline quoridor-app/app/api/game_api.js]: We don't name the file with `_`, just call it `game.js`

File: quoridor-app/app/api/game_api.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      body: JSON.stringify({
+        boardSize: 9,
+        numberOfFencesPerPlayer: 10
+      })
+    });
+    return await res.json();
+  }
+
+  async joinGame(gameId) {
+    const res = await fetch(API_PATH + "/" + gameId + "/join", {
```
Comment (zyhou) [inline quoridor-app/app/api/game_api.js]: use es6 style https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

File: quoridor-app/app/component/__tests__/pawn-test.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import renderer from "react-test-renderer";
+import { Text, StyleSheet } from "react-native";
+
+import { Pawn } from "../pawn";
+
+it("renders correctly", () => {
```
Comment (zyhou) [inline quoridor-app/app/component/__tests__/pawn-test.js]: add the `it` inside `describe`

Comment (zyhou) [reply quoridor-app/app/component/__tests__/pawn-test.js]: name like `it("should return the right value")`

File: quoridor-app/app/component/board.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+const styles = StyleSheet.create({
+  board: {
+    width: 360,
+    height: 360,
+    backgroundColor: "rgb(133, 65, 52)"
+  }
+});
+
+export class Board extends React.Component {
```
Comment (zyhou) [inline quoridor-app/app/component/board.js]: use functionnal style

File: quoridor-app/app/component/board.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  render() {
+    if (!this.props.squares) {
```
Comment (zyhou) [inline quoridor-app/app/component/board.js]: spread the props `const { squares } = this.props`

File: quoridor-app/app/component/board.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+      return null;
+    }
+    const delta = 40;
```
Comment (zyhou) [inline quoridor-app/app/component/board.js]: put outside of the component

File: quoridor-app/app/component/pawn.js [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+import { Text, StyleSheet } from "react-native";
+
+const styles = StyleSheet.create({
+  pawn: {
+    position: "absolute",
+    fontSize: 30
+  }
+});
+
+export class Pawn extends React.Component {
```
Comment (zyhou) [inline quoridor-app/app/component/pawn.js]: add proptypes

File: quoridor-app/app/component/square.js [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+  }
+});
+
+export class Square extends React.Component {
+  render() {
+    return (
+      <View
+        style={[
+          styles.square,
+          {
```
Comment (zyhou) [inline quoridor-app/app/component/square.js]: you can spread, `{...this.props.position}`

File: quoridor-app/app/screen/game.js [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+  container: {
+    width: "100%",
+    height: "100%",
+    justifyContent: "center",
+    flexDirection: "column",
+    alignItems: "center"
+  }
+});
+
+export default class GameScreen extends React.Component {
```
Comment (zyhou) [inline quoridor-app/app/screen/game.js]: use functional style + React hook

File: quoridor-app/app/component/pawn.js [hunk 13]
```diff
@@ -8,21 +9,27 @@ const styles = StyleSheet.create({
...
+      ]}
+    >
+      &#9823;
+    </Text>
+  );
+};
+
+Pawn.prototype = {
+  color: PropTypes.string.isRequired,
+  top: PropTypes.number.isRequired,
```
Comment (zyhou) [inline quoridor-app/app/component/pawn.js]: I didn't see this props in the component

File: quoridor-app/app/component/square.js [hunk 14]
```diff
@@ -10,18 +11,23 @@ const styles = StyleSheet.create({
...
+          top: position.top,
+          left: position.left
+        }
+      ]}
+    />
+  );
+};
+
+Square.prototype = {
+  top: PropTypes.number.isRequired,
```
Comment (zyhou) [inline quoridor-app/app/component/square.js]: same

File: quoridor-app/app/api/gameApi.js [hunk 15]
```diff
@@ -0,0 +1,31 @@
+const API_PATH = "http://192.168.86.56:8383/games";
```
Comment (alexisjanvier) [inline quoridor-app/app/api/gameApi.js]: Couldn't you use an environment variable to define this constant? (https://www.npmjs.com/package/babel-plugin-transform-inline-environment-variables)

File: quoridor-app/app/api/gameApi.js [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+    headers: {
+      Accept: JSON_TYPE,
+      "Content-Type": JSON_TYPE
+    },
+    body: JSON.stringify({
+      boardSize: DEFAULT_BOARD_SIZE,
+      numberOfFencesPerPlayer: DEFAULT_NUMBER_OF_FENCES_PER_PLAYER
+    })
+  });
+  return await response.json();
```
Comment (alexisjanvier) [inline quoridor-app/app/api/gameApi.js]: What happens if there is a call error?

File: quoridor-app/app/component/board.js [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+const Board = ({ squares, pawns }) => {
+  if (!squares) {
+    return null;
+  }
+
+  return (
+    <View style={styles.board}>
+      {squares.map((item, key) => (
+        <Square
+          key={key}
```
Comment (alexisjanvier) [inline quoridor-app/app/component/board.js]: it seems to me that it is not a good practice to use the map `index` as a `key` ... https://menubar.io/react-keys-index. For example, from line 30, you will generate identical keys

File: quoridor-app/app/component/pawn.js (hunk 13 above)
Comment (alexisjanvier) [reply quoridor-app/app/component/pawn.js]: nope, it's a `position` props.

File: quoridor-app/app/screen/game.js [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+        .then(game => {
+          setGame(game);
+          setIsLoading(false);
+        })
+        .catch(err => {
+          console.log({ err });
+          setIsLoading(false);
+        });
+    };
+    fetchGame();
```
Comment (alexisjanvier) [inline quoridor-app/app/screen/game.js]: missing the `return`? Or a condition of execution ?

File: README.md [hunk 19]
```diff
@@ -1,18 +1,20 @@
...
+> Expo cli should be installed globally: npm install expo-cli --global
 
 ```
 make install
 ```
 
 ## Start
 
-Run the application
+Run the application. Before starrting the application set your environment variables in ./quoridor-app/environment.js
```
Comment (alexisjanvier) [inline README.md]: s/starrting/starting/

File: quoridor-app/environment.js [hunk 20]
```diff
@@ -2,7 +2,7 @@ import { Constants } from "expo";
 
 const ENV = {
   dev: {
-    API_BASE_URL: "[your.dev.api.here]"
+    API_BASE_URL: "http://192.168.86.56:8383"
```
Comment (alexisjanvier) [inline quoridor-app/environment.js]: IMO, this ip is specific to your local environment, isn't it ? You should create an `environment.js.dist` with fake but explicit values, exlude this `environment.js` from git and add something about that in the `README` file.

File: quoridor-app/app/component/__tests__/board-test.js [hunk 21]
```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import renderer from "react-test-renderer";
+
+import Board from "../board";
+import Pawn from "../pawn";
+import Square from "../square";
+
+describe("Board Componenet tests", () => {
```
Comment (alexisjanvier) [inline quoridor-app/app/component/__tests__/board-test.js]: s/Componenet/Component/

File: quoridor-app/environment.js (hunk 20 above)
Comment (zyhou) [reply quoridor-app/environment.js]: If you follow Alexis advice about the `.dist`. You can add `cp -n environment.js.dist environment.js` in your `make install`

Comment (mchaffotte) [reply quoridor-app/environment.js]: environment.js is imported in gameApi so  cannot do in that way

Comment (alexisjanvier) [reply quoridor-app/environment.js]: The idea is to exlude the `environment.js` file from the git versioning in order not to risk exposing *secret* parameters But as Maxime explains, you have to copy an `environment.dist` as `environment.js` during the installation phase, and explain to your collaborators how to set the *adhoc* configuration in it in the `README` file.

File: quoridor-app/package.json [hunk 22]
```diff
@@ -20,6 +21,7 @@
     "react-navigation": "3.11.0"
   },
   "devDependencies": {
+    "@testing-library/react-native": "^4.0.7",
```
Comment (alexisjanvier) [inline quoridor-app/package.json]: It is better to stay on an exact version, i.e. to remain consistent with other dependencies. With npm, you can use the `--save-exact` option.

**Pull request merged**